pkg/auth/requests/authenticate.go

```diff
@@ -81,31 +92,37 @@ func tokenKeyIndexer(obj interface{}) ([]string, error) {
 	return []string{token.Token}, nil
 }
 
-func (a *tokenAuthenticator) Authenticate(req *http.Request) (bool, string, []string, error) {
+func (a *tokenAuthenticator) Authenticate(req *http.Request) (*AuthenticatorResponse, error) {
+	authResp := &AuthenticatorResponse{
+		IsAuthed: false,
+		User: "",
+		UserPrincipal: "",
+		Groups: []string{},
+	}
```

Comment:
This is initializing all the fields to the zero value, which we don't need to do, but it should be done for the map. A nil map will cause a panic if it is accessed.